java/code/src/com/redhat/rhn/domain/server/Server.java

```diff
@@ -2183,4 +2187,21 @@ public String getChannelHost() {
         return this.getFirstServerPath().map(p -> p.getHostname())
                 .orElseGet(() -> ConfigDefaults.get().getCobblerHost());
     }
+
+    public boolean checkIfInMaintenanceMode() {
+        if (maintenanceSchedule == null) {
+            return true;
+        }
+        if (!maintenanceSchedule.getCalendarOpt().isEmpty()) {
+            MaintenanceManager mm = MaintenanceManager.instance();
+            Collection<CalendarComponent> cc = mm.getDateFromSchedule(this.maintenanceSchedule);
+            if (cc.isEmpty()) {
+              return false;
+            }
+            else {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Move the whole logic  to `MaintenanceManager` (create a new method `boolean isServerInMaintenanceMode(server)` and call it with this). This will be clearer. There is no reason the `Server` class should do anything with `CalendarComponent`.

As for the implementation of the method itself:
1. fix the formatting
2. this 
```
            if (cc.isEmpty()) {
              return false;
            }
            else {
                return true;
            }
```
can be replaced with
```
return !cc.isEmpty();
```